app/src/main/java/com/example/cleanarchitechture/presentation/ui/MainFragment.kt

```diff
@@ -128,47 +130,44 @@ class MainFragment : Fragment(), ItemClickListener {
         super.onStart()
 
         requireContext().registerReceiver(
-            personAddedReceiver,
-            IntentFilter(Constants.PERSON_ADDED_BROADCAST)
+                personAddedReceiver,
+                IntentFilter(Constants.PERSON_ADDED_BROADCAST)
         )
         requireContext().registerReceiver(
-            batteryLeverBroadcastReceiver,
-            IntentFilter(Intent.ACTION_BATTERY_CHANGED)
+                batteryLeverBroadcastReceiver,
+                IntentFilter(Intent.ACTION_BATTERY_CHANGED)
         )
         sensorManager.registerListener(
-            accelerometerListener,
-            sensorGravity,
-            SensorManager.SENSOR_DELAY_NORMAL
+                accelerometerListener,
+                sensorGravity,
+                SensorManager.SENSOR_DELAY_NORMAL
         )
         if (ActivityCompat.checkSelfPermission(
-                requireContext(),
-                Manifest.permission.ACCESS_FINE_LOCATION
-            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
-                requireContext(),
-                Manifest.permission.ACCESS_COARSE_LOCATION
-            ) != PackageManager.PERMISSION_GRANTED
+                        requireContext(),
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                        requireContext(),
+                        Manifest.permission.ACCESS_COARSE_LOCATION
+                ) != PackageManager.PERMISSION_GRANTED
         ) {
-            // TODO: Consider calling
-            //    ActivityCompat#requestPermissions
-            // here to request the missing permissions, and then overriding
-            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
-            //                                          int[] grantResults)
-            // to handle the case where the user grants the permission. See the documentation
-            // for ActivityCompat#requestPermissions for more details.
-            return
+            requestPermissions(
+                    arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
```

Comment:
It's not a critical problem, but just a moment to think forward.
You are checking for any of 'fine' or 'coarse' location permission, but asking only for 'fine'. Potentially, could be misunderstanding in app behaviour and location accuracy result. 'Fine' means very accurate location. But if user grants 'coarse' location in system settings, you'll have not accurate location, but may consider that you're getting very accurate location. So, for me it's reasonable to ask and check same types of location.